### PR TITLE
Make CryptoHandler use Key as the abstraction layer

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## :scroll: Description

Updates `CryptoHandler` to use the new `Key` type as its key abstraction.

## :bulb: Motivation and Context

We're moving away from open-coded `String`s being used as keys.

## :green_heart: How did you test it?

Unit tests still pass.

## :pencil: Checklist

- [x] I formatted the code `./gradlew spotlessApply`
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.bouncycastle.openpgp.PGPKeyRing
+import org.pgpainless.PGPainless
+
+/** Utility methods to deal with PGP [Key]s. */
+public object KeyUtils {
+  /**
+   * Attempts to parse a [PGPKeyRing] from a given [key]. The key is first tried as a secret key and
+   * then as a public one before the method gives up and returns null.
+   */
+  public fun tryParseKeyring(key: Key): PGPKeyRing? {
```
Comment (Skrilltrax) [inline crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt:20]: Should we return a `Result<PGPKeyRing>` here?

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      return secKeyRing
+    }
+    val pubKeyRing = runCatching { PGPainless.readKeyRing().publicKeyRing(key.contents) }.get()
+    if (pubKeyRing != null) {
+      return pubKeyRing
+    }
+    return null
+  }
+
+  /** Parses a [PGPKeyRing] from the given [key] and returns its hex-formatted key ID. */
```
Comment (Skrilltrax) [inline crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt:32]: I'm still confused if we should make the API return nullable instance or should we just embrace `Result<T>`

File: crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt (hunk 1 above)
Comment (msfjarvis) [reply crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt:20]: I thought about it but I don't feel there's any value to throwing an exception here.

**Comment by Skrilltrax** [comment]
@msfjarvis overall this looks good, I just have one question though. Now, we do not have strict types for each key, and we are also not checking if the key passed is of the correct type. For example, someone can pass `AgeKey` to `PGPKeyManager`. The decryption will fail, but should we have runtime checks for `Key` type?

**Comment by msfjarvis** [comment]
> @msfjarvis overall this looks good, I just have one question though. Now, we do not have strict types for each key, and we are also not checking if the key passed is of the correct type. For example, someone can pass `AgeKey` to `PGPKeyManager`. The decryption will fail, but should we have runtime checks for `Key` type?

Yeah I considered that would potentially become a problem which is why I have defensive parsing tests for [`addKey`](https://github.com/android-password-store/Android-Password-Store/blob/ccb33af854132f1b35b71393ff68d24850de6960/crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt#L35) and [`removeKey`](https://github.com/android-password-store/Android-Password-Store/blob/ccb33af854132f1b35b71393ff68d24850de6960/crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPKeyManager.kt#L56). For now we'll just have to rely on whatever piece of code handles the handover from `PGPKeyManager` to `PGPainlessCryptoHandler` to not pass incorrect keys, but I'd like to enforce this on a type level in the future.

**Comment by Skrilltrax** [comment]
Cool, so we can have it as a TODO for when we add another backend

**Review by Skrilltrax — APPROVED** [review]

**Pull request merged**